Client/Pages/BookPages/BookDetails.razor

```diff
@@ -1,63 +1,105 @@
+                            <div>
+                                <NavLink class="button" href=@($"/authorView/{author.AuthorId}")>@author.FullName</NavLink> 
+                                <button type="button" class="button" @onclick="(() => DeleteAuthor(author.AuthorId))">Delete author</button>
+                        </div>
                         }
-                    </select>
-                </td>
-                <td>Authors</td>
-                <td><button @onclick="Edit">Edit book</button></td>
-            </tr>
-        </tbody>
-    </table>
+                    <div>
+                        <InputSelect @bind-Value="bookAuthor.AuthorId">
+                            @foreach (var author in allAuthors)
+                            {
+                                    @if (!author.Books.Any(b => b.BookId.Equals(id)))
+                                    {
+                                <option value="@author.AuthorId">@author.FullName</option>
+                                        }
+                                }
+                        </InputSelect>
+                        <button type="button" class="button" @onclick="AddAuthor">Add author</button>
+                    </div>
+                    </td>
+                    <td><button type="submit">Edit book</button></td>
+                </tr>
+            </tbody>
+
+        </table>
+    </EditForm>
+    
 }
 @code {
-    [Parameter]
+[Parameter]
     public Guid id { get; set; }
-    public Test.Shared.Book book { get; set; } = null;
-    private string nameToEdit;
-    private Guid publisherId;
+    public Test.Shared.Book book { get; set; } = new Test.Shared.Book();
+    private List<Test.Shared.Publisher> publishers = new List<Test.Shared.Publisher>();
+    private ICollection<Author> authors { get; set; } = new List<Author>();
+    public List<Author> allAuthors { get; set; } = new List<Author>();
+    public BookAuthor bookAuthor { get; set; } = new BookAuthor();
 
     private async Task Edit()
     {
-        book.BookName = nameToEdit;
-        book.BookPublisherId = publisherId;
-        await Http.PutAsJsonAsync<Test.Shared.Book>("book", book);
+        await Http.PutAsJsonAsync<Test.Shared.Book>($"api/books/{id}", book);
+        await OnInitializedAsync();
+    }
+
+    private async Task AddAuthor()
+    {
+        await Http.PostAsJsonAsync<BookAuthor>($"api/bookauthors", bookAuthor);
+        bookAuthor = new BookAuthor();
         await OnInitializedAsync();
     }
 
-    private void PublisherDetails(Guid id)
+    private async Task DeleteAuthor(Guid id)
     {
-        NavigationManager.NavigateTo("/publisherView/" + id);
+        foreach (var bookAuthor in book.BookAuthors)
+        {
+            if (bookAuthor.AuthorId == id)
+            {
+                await Http.DeleteAsync($"api/bookauthors/{bookAuthor.BookAuthorId}");
+                await OnInitializedAsync();
```

Comment:
Better check the return value and when Delete succeeds, remove the AuthorID from the local list by yourself. This triggers re-rendering.